### PR TITLE
[Feature] Assessment plan builder placeholder page

### LINKED ISSUE #8160: ✨  Assessment plan placeholder
## ✨ Feature

The assessment plan builder is the first step in the Record of decision (RoD) flow. It allows us to capture the list of assessments to be used during the staffing process, the order in which they will be performed, and the skills that each assessment will be targeting.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

1. Page makes database request to load assessment into memory
2. The title includes a chip/label to show the plan's current state (validation logic)
3. No copy for accordions yet - omit accordions for now
4. Omit card repeaters - will be added in "finalize" issue
5. Omit skill summary table - will be added in "finalize" issue
6. Build dialogs but don't make database request for mutations, launch from temporary buttons

## 🎨 Design File and Designer

[Link to figma file](https://www.figma.com/file/em5ArPr044YaIe1Vb1zCZz/Pools---Record-of-Decision?type=design&node-id=5437%3A34317&mode=design&t=11Dob6JX6BOb3lle-1) by @Jerryescandon 

## 📸 Screenshots

<details><summary> Full page screenshot</summary>
<p>

Frame 40

</p>
</details> 

<details><summary>Dialog (regular)</summary>
<p>

Assessment details - Repeater dialog 01

</p>
</details> 
<details><summary>Dialog (with screening questions) </summary>
<p>

Assessment details - Repeater dialog

</p>
</details> 

## ✅ Acceptance Criteria

- [ ] Page built with URL (like /en/admin/pools/:poolId/plan)
- [ ] Page meets all criteria from _Details_ section
- [ ] Hidden behind feature flag
- [ ] Design review with designer

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #8160

## 👋 Introduction

This branch creates a placeholder page for the assessment plan builder.  It queries for information, runs a validator, pops open dialogs, and (hopefully) gets all the copy translated.

## 🕵️ Details

Note the limitations of this issue: no card repeater, no skill table, no skill summary sidebar, no mutations run.  There is a lot of logic left commented out or stubbed to help finalize later.

~I guess the header that is locked to that path's children will be removed when the pages are finalized so I didn't bother wiring it up with a link or title.~  I checked with Jerbo and the header will stay.  However, the name of the page will be removed from the header.  Just the name of the pool will remain as the H1.  I'll update the "finalize page" issue.

## 🧪 Testing

1. Rebuild the app, ensure the record of decision flag is true
2. Log in as admin, navigate to `http://localhost:8000/en/admin/pools/:poolId/plan` with a pool that has pool skills defined
3. Check out main page with its copy and validator
4. Check out the dialogs - uncomment the console.debug in AssessmentDetailsDialog.tsx to see the form submission

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        {intl.formatMessage({
+          defaultMessage: "You are approaching the limit!",
+          id: "1moJ8r",
+          description:
+            "Title for warning message when the user has added many assessments to the assessment plan",
+        })}
+        {intl.formatMessage(
+          {
+            defaultMessage:
+              "You can add up to {maxSteps} assessment methods, but you don’t need to. You are covered as long as every skill has been taken into account. Choosing too many assessment methods can delay the staffing process and reduce the chances of a successful hire.",
```
Comment (esizer) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx:155]: Smart apostrophe on "don't" 🤓

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                  "On the initial application, only technical information is collected from the applicants, except for a few very special behavioural skills. This means that as the reviewer, when you see the application information, you’ll only have experience data related to the technical skills and a few very specific behavioural skills. This is done for two reasons.",
+                id: "3W5IAV",
+                description:
+                  "First paragraph of first answer of the Frequently Asked Questions for logging in",
+              })}
+            </p>
+            <p data-h2-margin-top="base(x.5)">
+              {intl.formatMessage({
+                defaultMessage:
+                  "Firstly, asking for too much information on a job application can drive away strong candidates who feel the process is too cumbersome. You’ll need to find a balance between how much you want to ask for and how long it will take an applicant to complete the process if you want the chance to recruit top candidates.",
```
Comment (esizer) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx:64]: Smart apostrophe on "you'll" 🤓

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                id: "HUU5sY",
+                description:
+                  "Second paragraph of first answer of the Frequently Asked Questions for logging in",
+              })}
+            </p>
+            <p data-h2-margin-top="base(x.5)">
+              {intl.formatMessage({
+                defaultMessage:
+                  "Secondly, after a year-long live pilot test, managers reported that they didn’t find the self-reported information on behavioural skills to be useful. Instead, they found other forms of assessment (such as reference checks and interviews) to be far better predictors than long paragraphs about how collaborative, positive or inclusive a candidate self-assessed as being.",
```
Comment (esizer) [inline apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx:73]: Smart apostrophe on "didn't" 🤓

File: packages/env/src/useFeatureFlags.ts [hunk 4]
```diff
@@ -1,10 +1,7 @@
 import { getFeatureFlags } from "./utils";
 
-export type FeatureFlags = {
-  skillLibrary: boolean;
-};
-
-const useFeatureFlags = (): FeatureFlags => {
+export type FeatureFlags = ReturnType<typeof getFeatureFlags>;
```
Comment (esizer) [inline packages/env/src/useFeatureFlags.ts:3]: Clever! 👏

**Review by esizer — CHANGES_REQUESTED** [review]
Looks good! Some very minor items to take a look at.

**Comment by Jerryescandon** [comment]
Reviewed with @petertgiles , looks good to me 👍

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx (hunk 1 above)
Comment (petertgiles) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/OrganizeSection.tsx]: Nice catch! [get rid of smart apostrophes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8225/commits/c36c915fdbdfda63a89042cb100959aed553ab6c)

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                description:
+                  "first question in the skill summary section in the assessment plan builder",
+              })}
+            </Accordion.Trigger>
+          </Accordion.Header>
+          <Accordion.Content>
+            <p>
+              {intl.formatMessage({
+                defaultMessage:
+                  "On the initial application, only technical information is collected from the applicants, except for a few very special behavioural skills. This means that as the reviewer, when you see the application information, you’ll only have experience data related to the technical skills and a few very specific behavioural skills. This is done for two reasons.",
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx]: Nice catch! [get rid of smart apostrophes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8225/commits/c36c915fdbdfda63a89042cb100959aed553ab6c)

File: apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx (hunk 3 above)
Comment (petertgiles) [reply apps/web/src/pages/Pools/AssessmentPlanBuilderPage/components/SkillSummarySection.tsx]: Nice catch! [get rid of smart apostrophes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8225/commits/c36c915fdbdfda63a89042cb100959aed553ab6c)

**Review by esizer — APPROVED** [review]
LGTM! 👍

**Pull request merged**